src/mono/mono/metadata/assembly.c

```diff
@@ -1642,6 +1642,27 @@ load_reference_by_aname_individual_asmctx (MonoAssemblyName *aname, MonoAssembly
 	return reference;
 }
 #else
+static MonoAssembly *
+search_bundle_for_assembly (MonoAssemblyLoadContext *alc, MonoAssemblyName *aname)
+{
+	if (bundles == NULL)
+		return NULL;
+
+	MonoImageOpenStatus status;
+	MonoImage *image;
+	MonoAssemblyLoadRequest req;
+	image = mono_assembly_open_from_bundle (alc, aname->name, &status, FALSE);
+	if (!image) {
+		char *name = g_strdup_printf ("%s.dll", aname->name);
```

Comment:
Does `mono_assembly_open_from_bundle` work if `aname->name` doesn't have a `.dll` suffix? If it only works with `.dll` suffix then would it make sense to default to appending `.dll` if it doesn't have it so we would only call `mono_assembly_open_from_bundle` once?

Reply:
It totally depends on how the name is registered in the bundle.

Our behavior has historically been to just try with the name directly and then, if it fails, try postfixing, which seems sane to me. You have to try both cases unless we want to break historical behavior (we don't), and trying the original name first seems optimal.

Worth noting I haven't changed this logic at all in the PR, just moved where it occurs.